wrappers/python/src/cxx/api/ApiModule.cpp

```diff
@@ -159,6 +160,42 @@ void exportApi(py::module &api) {
             }, "n_steps"_a, "time_step"_a);
     exportObservables(api, simulation);
 
+    // actions and evaluate observables, i.e. things needed to build a custom simulation loop [experimental]
+    {
+        using Action = readdy::model::actions::Action;
+        using EvalObs = readdy::model::actions::EvaluateObservables;
+        using MkCkpt = readdy::model::actions::MakeCheckpoint;
+
+        auto actionsModule = api.def_submodule("actions");
+        py::class_<Action>(actionsModule, "Action")
+                .def("__call__", &Action::perform);
+
+        py::class_<readdy::model::actions::top::BreakConfig>(actionsModule, "BreakConfig")
+                .def(py::init<>())
+                .def("add_breakable_pair", &readdy::model::actions::top::BreakConfig::addBreakablePair);
+
+        simulation
+        .def("create_action_initialize_kernel", [](sim &self) -> std::unique_ptr<Action> { return self.actions().initializeKernel(); })
+        .def("create_action_euler_bd", [](sim &self, readdy::scalar timeStep) -> std::unique_ptr<Action> { return self.actions().eulerBDIntegrator(timeStep); })
+        .def("create_action_calculate_forces", [](sim &self) -> std::unique_ptr<Action> { return self.actions().calculateForces();})
+        .def("create_action_create_neighbor_list", [](sim &self, readdy::scalar interactionDistance) -> std::unique_ptr<Action> { return self.actions().createNeighborList(interactionDistance);})
+        .def("create_action_update_neighbor_list", [](sim &self) -> std::unique_ptr<Action> { return self.actions().updateNeighborList();})
+        .def("create_action_clear_neighbor_list", [](sim &self) -> std::unique_ptr<Action> { return self.actions().clearNeighborList();})
+        .def("create_action_uncontrolled_approximation", [](sim &self, readdy::scalar timeStep) -> std::unique_ptr<Action> { return self.actions().uncontrolledApproximation(timeStep);})
+        .def("create_action_gillespie", [](sim &self, readdy::scalar timeStep) -> std::unique_ptr<Action> { return self.actions().gillespie(timeStep);})
+        .def("create_action_detailed_balance", [](sim &self, readdy::scalar timeStep) -> std::unique_ptr<Action> { return self.actions().detailedBalance(timeStep);})
+        .def("create_action_evaluate_topology_reactions", [](sim &self, readdy::scalar timeStep) -> std::unique_ptr<Action> { return self.actions().evaluateTopologyReactions(timeStep);})
+        .def("create_action_break_bonds", [](sim &self, readdy::scalar timeStep, const readdy::model::actions::top::BreakConfig &breakConfig) -> std::unique_ptr<Action> { return self.actions().breakBonds(timeStep, breakConfig);});
+
+        // strictly not an action
```

Comment:
or is it?

Reply:
actually I would advocate offering both. The action interface offers `perform()` which dispatches to a secondary method `perform(t)`.